src/Microsoft.AspNet.Hosting/Internal/WebApplication.cs

```diff
@@ -98,14 +127,14 @@ public virtual IApplication Start()
             _applicationLifetime.NotifyStarted();
             logger.Started();
 
-            return new Application(ApplicationServices, Server.Features, new Disposable(() =>
-            {
-                logger.Shutdown();
-                _applicationLifetime.StopApplication();
-                Server.Dispose();
-                _applicationLifetime.NotifyStopped();
-                (_applicationServices as IDisposable)?.Dispose();
-            }));
+            return new Disposable(() =>
```

Comment:
This would not cover the cases where we want to configure things like addresses after the application builds but before the application runs. Also, doesn't `Start()` or `Run()` feel a little out of place as an extension to a `Builder`?